operator/v1/types_ingress.go

```diff
@@ -665,6 +674,21 @@ type IBMLoadBalancerParameters struct {
 	Protocol IngressControllerProtocol `json:"protocol,omitempty"`
 }
 
+// OpenStackLoadBalancerParameters provides configuration settings that are
+// specific to OpenStack load balancers.
+type OpenStackLoadBalancerParameters struct {
+	// loadBalancerIP specifies the IP address that the load balancer will use.
+	// Note: this field is meant to be set by the ingress controller to populate the
+	// `Service.Spec.LoadBalancerIP` field which has been deprecated in Kubernetes:
+	// https://github.com/kubernetes/kubernetes/pull/107235
+	// However, the field is still used by cloud-provider-openstack to reconcile
+	// the floating IP that we attach to the load balancer.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	LoadBalancerIP string `json:"loadBalancerIP,omitempty"`
```

Comment:
Lets make this a *string. Also, I heard a rumour that we can validate this as an IP address (v4 or v6). I suspect @JoelSpeed knows the incantation.

Reply:
Doesn't need to be a pointer does it? Omitempty should suffice and you can just check for length?

If you want to check this is a valid IP, try
```
// +kubebuilder:validation:XValidation:rule="isIP(self)",message="loadBalancerIP must be a valid IPv4 or IPv6 address"
```
Do you need any complex validation apart from it being an IP?

Reply:
True, combined with IP validation.